backend/src/controller/signup-controller.js

```diff
@@ -0,0 +1,37 @@
+import { validate } from "class-validator";
+import { BadRequestError } from "../common/error/bad-request-error";
+import { UserService } from "../service";
+import { EntityAlreadyExist } from "../common/error/entity-already-exist";
+
+const validateSignupParam = async (req, res, next) => {
+    const { email, name, password } = req.body;
+    const userService = new UserService();
```

Comment:
```suggestion
    const userService = UserService.getInstance();
```
UserService를 싱글톤으로 가져가면 매번 UserService 인스턴스를 만들지 않아도 될 것 같아요!

Reply:
@youngxpepp 함수형으로 구현하다보니 매번 서비스 인스턴스 생성에 대해 고민중이었는데 싱글톤 방식이 해결책이 될 수 있겠군요!
제안해주신 코드 잘 참고하였습니다 :-) 평소 싱글톤 객체를 만들때 모듈을 이용하고 있습니다. Node.js로 모듈파일 내에서 인스턴스 생성 후 내보내면 캐시가 적용되어 해당 모듈을 참조하는 곳에서는 같은 인스턴스를 참조하여 싱글톤을 적용할 수 있게 되는데 제안해주신 코드처럼 패턴을 직접 구현하는 것과 차이점이 존재할까요??

Reply:
차이가 없을 듯 해요ㅎㅎㅎ 클래스로 선언된 만큼 최대한 객체지향적으로 설계하려고 했었는데, 우진님 방식이 더 간단하다고 생각이 듭니다!

Reply:
@youngxpepp 그렇군요! 모듈 방식을 이용하면 코드는 간결해지지만,  Class에 패턴을 구현하면, 확실히 해당 서비스를 참조해서 개발하시는 다른 분들이 싱글톤 패턴을 구현했는지 명확하게 이해할 것 같습니다 :-)